theme/static/js/hs_file_browser.js

```diff
@@ -2322,6 +2322,28 @@ $(document).ready(function () {
         return false;
     });
 
+    // User clicked Proceed button on confirm whether to override files dialog when unzipping - need to override files
+    $("#btn-unzip-file-override-proceed").click(function () {
+        var calls = [];
+        var res_id = $("#res_id").val();
+        var zip_with_rel_path = $("#zip_with_rel_path").val();
+        calls.push(unzip_irods_file_ajax_submit(res_id, zip_with_rel_path, overwrite='true'));
+        // Disable the Cancel button until request has finished
+        $(this).parent().find(".btn[data-dismiss='modal']").addClass("disabled");
+        function afterDoneRequest() {
+            $('#btn-unzip-file-override-proceed').parent().find(".btn[data-dismiss='modal']").removeClass("disabled");
+            $("#unzip-override-confirm-dialog").modal('hide');
+        }
+        function afterFailRequest() {
+            afterDoneRequest();
+            refreshFileBrowser();
+        }
+
+        $.when.apply($, calls).done(afterDoneRequest);
```

Comment:
I've only ever seen deferred methods to jQuery.when() called using method chaining as in
```
$.when.apply($, calls).done(afterDoneRequest).fail(afterFailRequest);
```
https://api.jquery.com/deferred.fail/
Does it actually work the way you've written it here? If so, that is good learning for me...I thought they had to be chained.

Reply:
Based on my testing, it works as expected. I tried not to refresh browser when the async call is done successfully since otherwise, the temporary folder where the unzipped files are put would show up in the file browser. However, when the async call failed, refreshFileBrowser will be called. It appears to work as expected since before I made this change, the temporary intermediate folder showed up in the file browser, and after I made the change, the temporary folder does not show up any more. It is good to know the chaining rule for deferred methods to JQuery.when(), but it appears not chaining also works. Just did a search in the file and found the chaining is not used in other places as well, e.g., [here](https://github.com/hydroshare/hydroshare/blob/develop/theme/static/js/hs_file_browser.js#L631-L642).

Reply:
nice! thanks for the learning